### DISCUSSION TITLE
React Hook useEffect usability.

### DISCUSSION THREAD

**Discussion by miracle0315** [community/community#19952] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi !
I know that useEffect function on React Hook have 2 parameters.
Like this.
    useEffect(() => {
        if (auth.isAuthenticated) 
            getPosts();
         else
            getBlogs();
    }, [getPosts]);
Here getPosts is action calling.
Then what is the role of second parameter???
<img width="688" alt="Screenshot 2022-04-08 194054" src="https://user-images.githubusercontent.com/104805337/177073339-86a1a996-ec71-4178-9d43-1aa3b1f71d5a.png">

**Comment by naji0329** [comment, marked as answer]
useEffect is a function which takes two arguments. The first argument passed to useEffect is a function called effect (You can guess why this hook is named useEffect) and the second argument (which is optional) is an array of dependencies. Below is an illustration of how it is used.

**Reply by miracle0315** [reply]
You are right!
